core/src/main/java/org/apache/iceberg/io/OutputFileFactory.java

```diff
@@ -158,12 +161,24 @@ public Builder suffix(String newSuffix) {
       return this;
     }
 
+    public Builder ioSupplier(Supplier<FileIO> newIoSupplier) {
```

Comment:
nit: add a Javadoc on why this is a supplier for dynamic refreshing capability

Reply:
Thanks, I added this.